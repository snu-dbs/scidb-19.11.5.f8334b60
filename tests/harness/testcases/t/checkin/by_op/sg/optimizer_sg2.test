--setup
--start-query-logging

create array S <val : double> [x=0:1,1,0]

# obfuscated way to set x[0]=2, x[1]=0
store(build(S, 2-(x*2)),S)
scan(S)  -- Inserted by scan_doctor

# expected output
#{x} val
#{0} 2
#{1} 0


--test

subarray(S,0,1)
# expected output
#{x} val
#{0} 2
#{1} 0

subarray(S,1,2)
# expected output
#{x} val
#{0} 0

join(subarray(S,0,1) as A, subarray(S,1,2) as B)
# expected output
#{x} A.val, B.val
#{0} 2,     0

apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val-B.val)
# expected output
#{x} A.val, B.val, c
#{0} 2,     0,     2

project(apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val-B.val),c)
# expected output
#{x} c
#{0} 2


# store(
#       project(apply(join(
#                          subarray(S,0,1) as A, #  same as S, x=[0:1] val <2,0>
#                          subarray(S,1,2) as B) #  reduced to x=1:1, becomes x=[0:0] val <0>
#                          # result of join is so [0:0] <A.val=2, B.val=0>
#                      c,                    #  so c    <2-0> = <2>
#                      A.val - B.val)  #
#                c),
#       R)
#
store(project(apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val - B.val), c), R)
scan(R)  -- Inserted by scan_doctor

#expected outputs
#{x} c
#{0} 2

--cleanup
remove(S)
remove(R)
