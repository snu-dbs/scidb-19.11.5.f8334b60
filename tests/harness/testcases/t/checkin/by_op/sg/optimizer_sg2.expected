SCIDB QUERY : <create array S <val : double> [x=0:1,1,0]>
Query was executed successfully

SCIDB QUERY : <store(build(S, 2-(x*2)),S)>
Query was executed successfully

SCIDB QUERY : <scan(S)  -- Inserted by scan_doctor>
{x} val
{0} 2
{1} 0

SCIDB QUERY : <subarray(S,0,1)>
{x} val
{0} 2
{1} 0

SCIDB QUERY : <subarray(S,1,2)>
{x} val
{0} 0

SCIDB QUERY : <join(subarray(S,0,1) as A, subarray(S,1,2) as B)>
{x} val,val
{0} 2,0

SCIDB QUERY : <apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val-B.val)>
{x} val,val,c
{0} 2,0,2

SCIDB QUERY : <project(apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val-B.val),c)>
{x} c
{0} 2

SCIDB QUERY : <store(project(apply(join(subarray(S,0,1) as A, subarray(S,1,2) as B), c, A.val - B.val), c), R)>
Query was executed successfully

SCIDB QUERY : <scan(R)  -- Inserted by scan_doctor>
{x} c
{0} 2

SCIDB QUERY : <remove(S)>
Query was executed successfully

SCIDB QUERY : <remove(R)>
Query was executed successfully

